Use pkg-config to link against lua

**Please, do not create duplicate issues**


### Actual behaviour

Compilation fails on master with:
```
/home/teto/USDX/src/media/UAudioDecoder_FFmpeg.pas(325,3) Note: Local variable "TestFrame" not used
Compiling media/UMedia_dummy.pas
Compiling menu/UMenuStaticList.pas
Linking ../game/ultrastardx
Warning: "crti.o" not found, this will probably cause a linking failure
Warning: "crtn.o" not found, this will probably cause a linking failure
/nix/store/2dfjlvp38xzkyylwpavnh61azi0d168b-binutils-2.31.1/bin/ld : avertissement : ../game/link.res contient des sections de sortie; avez-vous oublié -T?
/nix/store/2dfjlvp38xzkyylwpavnh61azi0d168b-binutils-2.31.1/bin/ld : ne peut trouver -llua5.2
Error: Error while linking
Fatal: There were 1 errors compiling module, stopping
Fatal: Compilation aborted
Error: /nix/store/6y49zi0s1s7fwirv4yd3835khnmh0wl0-fpc-3.0.0/bin/ppcx64 returned an error exitcode
make[1]: *** [Makefile:233: ../game/ultrastardx] Error 1
make[1] : on quitte le répertoire « /home/teto/USDX/src »
make: *** [Makefile:115: all] Error 2
```
This is the result of the configure
```
configure:4913: checking for lua5.3
configure:4921: $PKG_CONFIG --exists --print-errors "lua5.3"
Package lua5.3 was not found in the pkg-config search path.
Perhaps you should add the directory containing `lua5.3.pc'
to the PKG_CONFIG_PATH environment variable
No package 'lua5.3' found
configure:4924: $? = 1
configure:4952: result: no
configure:5056: checking for lua5.2
configure:5064: $PKG_CONFIG --exists --print-errors "lua5.2"
configure:5067: $? = 0
configure:5087: result: yes (-L/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/lib)
configure:5119: checking version of lua
configure:5128: $PKG_CONFIG --exists --print-errors "lua5.2"
configure:5131: $? = 0
configure:5154: result: [5.2.4]
configure:5485: checking size of lua_Integer
configure:5494: $PKG_CONFIG --exists --print-errors "$lua_LIB_NAME"
configure:5497: $? = 0
configure:5523: gcc -o conftest -g -O2 -I/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/include   conftest.c  -L/nix/store/5rhyabiwij29b3lqmgxzx5fiqcrwjkcd-SDL2-2.0.9/lib -L/nix/store/5rhyabiwij29b3lqmgxzx5fiqcrwjkcd-SDL2-2.0.9/lib -L/nix/store/c15bk2sc3djw3w4fzdkbb3l2hahkh5b8-SDL2_image-2.0.4/lib -L/nix/store/zxdnpb673bf27hcqzvssv5629m4x5bjv-freetype-2.10.0/lib -L/nix/store/ap4sr1n0wlgmybxbw3pvq8klh8snc3n8-sqlite-3.28.0/lib -L/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/lib >&5
configure:5523: $? = 0
configure:5523: ./conftest
configure:5523: $? = 0
configure:5529: result: 64 bits
```


### Expected behaviour

I believe rather than guessing what to link, if pkg-config is available then the softwaure should just use pkg-config --libs lua to get the LDFLAGS. For instance on my system, the content of lua5.2.pc is:
```
prefix=/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4
libdir=/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/lib
includedir=/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/include
INSTALL_BIN=/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/bin
INSTALL_INC=/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/include
INSTALL_LIB=/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/lib
INSTALL_MAN=/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/man/man1

Name: Lua
Description: An Extensible Extension Language
Version: 5.2.4
Requires:
Libs: -L/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/lib -llua -lm
Cflags: -I/nix/store/05hwxyizx7bc4p01jgf21vwlk2k7prri-lua-5.2.4/include
```


- USDX version: master 8d9b3e32891983f6d128d90c68e4e364c01b2e11
- Operating System + version: nixos-unstable (www.nixos.org)

--- a/configure.ac
+++ b/configure.ac
@@ -227,7 +227,6 @@ else
 		fi
 	fi
 fi
-AC_SUBST(lua_LIB_NAME)
 AC_MSG_CHECKING([size of lua_Integer])
 PKG_VALUE([lua],[CFLAGS],[cflags],[$lua_LIB_NAME],[C compiler flags for Lua])
 old_CFLAGS="$CFLAGS"
@@ -236,6 +235,16 @@ AC_COMPUTE_INT([LUA_INTEGER_BITS],[(long)sizeof(lua_Integer)*8],[#include <lua.h
 CFLAGS="$old_CFLAGS"
 AC_SUBST(LUA_INTEGER_BITS)
 AC_MSG_RESULT([$LUA_INTEGER_BITS bits])
+lua_lib_candidates=
+for i in $lua_LIBS ; do
+	case "$i" in
+	-l*) lua_lib_candidates="$lua_lib_candidates ${i#-l}"
+	esac
+done
+old_LIBS=$LIBS
+AC_SEARCH_LIBS([luaL_newstate], [$lua_lib_candidates], [lua_LIB_NAME=${ac_cv_search_luaL_newstate#-l}], [], [])
+LIBS=$old_LIBS
+AC_SUBST(lua_LIB_NAME)
 
 # find FFMpeg
 # Note: do not use the min/max version parameters with ffmpeg

